src/main/java/carbon/Parser.java

```diff
@@ -0,0 +1,97 @@
+package carbon;
+
+import carbon.error.CarbonException;
+import carbon.error.CorruptedSavefileException;
+import carbon.error.InvalidInputException;
+import carbon.task.Task;
+
+/**
+ * Parses user inputs, and processes them.
+ * This class merges the reading and writing functionality of the chat bot, by
+ * parsing user inputs, and calling for the necessary processing to be executed.
+ */
+public class Parser {
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructs an instance of Parser class.
+     *
+     * @param ui An instance of the Ui class.
+     * @param storage An instance of the Storage class.
+     * @return Parser object.
+     */
+    public Parser(Ui ui, Storage storage) {
+        this.ui = ui;
+        this.storage = storage;
+        try {
+            this.taskList = this.storage.loadSavefile();
+        } catch (CorruptedSavefileException error) {
+            // continue with empty list
+        }
+    }
+
+    /**
+     * Parses and processes a user input text.
+     * If the command is not a basic command, this method will call the processAdvanced method.
+     *
+     * @param input User input text.
+     * @return Execution log.
+     * @throws CarbonException  If an error is encountered during processing or execution.
+     */
+    public String process(String input) throws CarbonException {
```

Comment:
Great use of switch statements to improve the readability of the code. Well done!